api_loader.go

```diff
@@ -297,6 +297,11 @@ func processSpec(spec *APISpec, apisByListen map[string]int,
 		mwAppendEnabled(&chainArray, &VersionCheck{BaseMiddleware: baseMid})
 		mwAppendEnabled(&chainArray, &RequestSizeLimitMiddleware{baseMid})
 		mwAppendEnabled(&chainArray, &TrackEndpointMiddleware{baseMid})
+
+		if spec.UseMutualTLSAuth {
```

Comment:
define `EnabledForSpec` instead

Reply:
I'm not sure what you mean here if you look at the whole function this is a standard way how we conditionally include middleware: `if spec.UseBasicAuth`, `if spec.EnableSignatureChecking`, `if spec.EnableJWT` and etc.

Reply:
Then all of them are wrong :) all these should use `EnabledForSpec`, as per the interface contract.

This was probably done this way because the contract wasn't used properly before, which I fixed recently. Feel free to leave it like this though, as it won't be the only one to fix.

Reply:
Yeah, will leave it for you to refactor one day :)

Reply:
Added to my ever growing list of stuff to clean up ;)